packages/health-check/src/checks/critical.js

```diff
@@ -60,7 +86,44 @@ async function uploadCheck(done) {
     data.ip = error?.response?.ip ?? null;
   }
 
-  done({ name: "upload_file", time: calculateElapsedTime(time), ...data });
+  done({ name, time: calculateElapsedTime(time), ...data });
+}
+
+// skylinkHealthCheck checks if the skylink has reached full redundancy
+async function skylinkHealthCheck(skylink, numRetries = 30, authCookie, isLarge = false) {
+  // Get the health of the skylink
+  const response = await got(`https://${process.env.PORTAL_DOMAIN}/skynet/health/skylink/${skylink}`, {
+    headers: { cookie: authCookie },
+  });
+  const healthData = getResponseContent(response);
+
+  // Check Basesectorredundancy first
+  if (healthData.basesectorredundancy !== 10 && numRetries > 0) {
+    // Semi-smart sleep before retrying. Sleep longer if the redundancy is
+    // lower.
+    await new Promise((r) => setTimeout(r, (10 - healthData.basesectorredundancy) * 1000));
+    return skylinkHealthCheck(skylink, numRetries - 1, authCookie, isLarge);
+  }
+
+  // Check the Fanout redundancy if it is a large file
+  if (isLarge && healthData.fanoutredundancy != 3 && numRetries > 0) {
+    // Semi-smart sleep before retrying. Sleep longer if the redundancy is
+    // lower.
+    await new Promise((r) => setTimeout(r, (3 - healthData.fanoutredundancy) * 10000));
```

Comment:
For the large file it is possible that this can take way more than 5 minutes with all the retries and also waiting for the endpoint to return skylink data - the idea for critical checks is that the checks are executed every 5 minutes and should be almost instant so we know how the system operates at the specific point of time and we can pull it out of load balancer right away if there's an issue. If we're considering that a skylink can take 5+ minutes to reach full redundancy and still succeed then we should move this to extended tests probably that run once an hour. Otherwise we should assume a shorter (< 1 minute) total time for the critical test suite to either fail or succeed.

Reply:
a 40MB file shouldn't take more than 5 minutes to upload. If it does, I would argue that the portal isn't operating as expected and should be pulled. 

but i think its find to move this to the extended tests.

Reply:
We can leave it in the critical test but we should set the timeout to something like 2 minutes. We should be able to set a time threshold for uploading a 40MB file directly from the server so with no network latency - a minimum performance threshold of some sort.